main.cpp

```diff
@@ -8,49 +8,58 @@
 #endif
 
 wgpu::Instance instance;
+wgpu::Adapter adapter;
 wgpu::Device device;
 wgpu::RenderPipeline pipeline;
 
-wgpu::SwapChain swapChain;
+wgpu::Surface surface;
 const uint32_t kWidth = 512;
 const uint32_t kHeight = 512;
 
-void SetupSwapChain(wgpu::Surface surface) {
-  wgpu::SwapChainDescriptor scDesc{
-      .usage = wgpu::TextureUsage::RenderAttachment,
-      .format = wgpu::TextureFormat::BGRA8Unorm,
+void ConfigureSurface() {
+  wgpu::SurfaceCapabilities capabilities;
+  surface.GetCapabilities(adapter, &capabilities);
+
+  wgpu::SurfaceConfiguration config{
+      .device = device,
       .width = kWidth,
       .height = kHeight,
-      .presentMode = wgpu::PresentMode::Fifo};
-  swapChain = device.CreateSwapChain(surface, &scDesc);
+      .format = capabilities.formats[0],
+      .alphaMode = capabilities.alphaModes[0],
+      .presentMode = capabilities.presentModes[0]};
+  surface.Configure(&config);
 }
 
-void GetDevice(void (*callback)(wgpu::Device)) {
+void GetAdapter(void (*callback)(wgpu::Adapter)) {
   instance.RequestAdapter(
       nullptr,
       // TODO(https://bugs.chromium.org/p/dawn/issues/detail?id=1892): Use
-      // wgpu::RequestAdapterStatus, wgpu::Adapter, and wgpu::Device.
+      // wgpu::RequestAdapterStatus and wgpu::Adapter.
       [](WGPURequestAdapterStatus status, WGPUAdapter cAdapter,
          const char* message, void* userdata) {
         if (status != WGPURequestAdapterStatus_Success) {
           exit(0);
```

Comment:
nit: print the `message`

Reply:
I think it makes it a bit more useful as a reference, because if someone tries to run the sample and it fails, they'll see some explanation of why. But no strong opinion.